cache/src/lib.rs

```diff
@@ -150,46 +155,156 @@ pub struct CacheItem {
 	// The accessbile index in parent.  i
 	pub index: i32,
 	// Child count of the accessible  i
-	pub children: i32,
+	pub children_num: i32,
 	// The exposed interfece(s) set.  as
-	pub ifaces: InterfaceSet,
+	pub interfaces: InterfaceSet,
 	// Accessible role. u
 	pub role: Role,
 	// The states applicable to the accessible.  au
 	pub states: StateSet,
 	// The text of the accessible.
 	pub text: String,
-}
-impl TryFrom<atspi::cache::CacheItem> for CacheItem {
-	type Error = AccessiblePrimitiveConversionError;
+	// The children (ids) of the accessible.
+	pub children: Vec<AccessiblePrimitive>,
 
-	fn try_from(atspi_cache_item: atspi::cache::CacheItem) -> Result<Self, Self::Error> {
+	#[serde(skip)]
+	pub cache: Weak<Cache>,
```

Comment:
On second thought, since you are using the reference to the cache from within the cache methods, it might create some footguns (i.e. all the dashmap cases where `Locking behaviour: May deadlock if called when holding any sort of reference into the map.`). Probably still possible to reduce to one level though.